independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -541,8 +541,6 @@ private static int compareAlternativeBeans(InjectableBean<?> bean1, InjectableBe
                 }
             }
         }
-        // Observers with smaller priority values are called first
-        Collections.sort(resolvedObservers, InjectableObserverMethod::compare);
```

Comment:
Why do you remove this line? We can sort them once and then it's not necessary to sort them every time a notifier is executed, or?

Reply:
Because I split them into tx and standard ones and then, when JTA is not available, I need to merge them together again, which means I need to sort them again anyway so this just avoids double sort.

Reply:
Hm, you mean when a transaction is not available? We could probably optimize this a little bit...

Reply:
In what way? We cannot really know before hand if a tx will be active (e.g. if we will need to merge them together).

Reply:
Well, you could only have one sorted list `List<InjectableObserverMethod<? super T>>` and when a TX is not active (or no TX observers) just use the list or when a TX is active simply filter the list and add only TX observers to the synchronization...